e2e/cypress/support/commands.ts

```diff
@@ -201,3 +253,24 @@ function LoginHelper(username: string) {
     cy.wait(['@getAuthPopulateCache', '@getAuthParameterized']);
   });
 }
+
+function waitUntilAdminTokenPermissioned(attempts: number): void {
+  if (attempts === -1) {
+    throw new Error('admin token failed to generate');
+  }
+  // admin-only API endpoint
+  cy.request({
+    headers: { 'api-token': Cypress.env('ADMIN_TOKEN') },
+    url: '/apis/iam/v2beta/projects',
+    method: 'GET',
+    failOnStatusCode: false
+  }).then((response) => {
+    if (response.status === 200) {
+      return;
+    } else {
+      cy.log(`${attempts} attempts remaining: waiting for admin token to be permissioned`);
+      cy.wait(1000);
+      waitUntilAdminTokenPermissioned(--attempts);
```

Comment:
You are just doing a simple loop here, so no need for recursive calls (which take a lot more overhead).